go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -974,13 +939,12 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 		}
 		if !isNewReady {
 			klog.Warningf("Node: %s is not ready, deleting it from egress assignment", newNode.Name)
-			if err := h.oc.deleteEgressNode(newNode.Name); err != nil {
+			if err := h.oc.deleteEgressNode(newNode); err != nil {
 				return err
 			}
-		} else if isNewReady && isNewReachable {
+		} else if isNewReady {
 			klog.Infof("Node: %s is ready and reachable, adding it for egress assignment", newNode.Name)
```

Comment:
we won't know if its reachable or not unfortunately :/ before ovnkube-master used to check for reachable, ready and egressLabels, this behaviour will change to only checking for ready and egressLabels. We will add nodeIP to the address set of the 102 default LRP and do the GARP setup even if node is not reachable, although if node becomes non-reachable and if that node is used by egressIP, cluster-manager will trigger a status update and we will take action and delete that. That is our only signal for reachability. This was done intentionally because maintaining the whole healthcheck portions across all masters seemed like an overkill just to maintain that logic here.

As for nodes that are not allocated, tough luck.

Reply:
@trozet and I spoke in depth about this and reached the conclusion that we don't really need any of the stuff we do in `reconcileNode`.
We will keep `GARP`, `addressSet` and `nodeZoneState` add/delete/update. no need for anything else..

Reply:
We talked on slack about removing most of reconcileNodeForEgressIP

Reply:
done.